### DISCUSSION TITLE
xDrip+ flagged by Anti Virus

### DISCUSSION THREAD

**Discussion by Puderzo** [NightscoutFoundation/xDrip#3315] [Bug reports]
The lastest xDrip (Nightly build 2db9aa6-2024.01.16) got flagged by the Anti Virus (Palo Alto Cortex XDR) on my phone yesterday. I have attached the verdict report for the devs to study, perhaps you'll find something here that you could change to make this not happen in the future.

[global-verdict-report (5).pdf](https://github.com/NightscoutFoundation/xDrip/files/14051030/global-verdict-report.5.pdf)

I expect this to be a False Positive, but would like a confirmation from the dev team before I flag this as such. I have been using both Cortex XDR and xDrip together for over a year now, and this is the first time it's got flagged, so something seems to have changed in the last Nightly that did this.

**Comment by Navid200** [comment]
Can you find a version of xDrip that does not cause this?
And can you find a version of xDrip that causes this right away when you install consistently?

**Reply by Puderzo** [reply]
Actually, I was on Alpha channel, not Nightly. The current Beta is safe (not flagged as malware) whereas when I change to alpha channel and update, it's flagged as malware. So 20231114-3e86af7 works but not 20240116-2db9aa6. I'm now testing with the previous Nightly 20240113-1af5128, which is currently pending verdict. If this is flagged, I will try to find the earliest version which is not flagged. Will keep you posted.

**Reply by Puderzo** [reply]
Ok, so the last working (not flagged) version is 20231121-c164980 and the first flagged version is 20231128-a8b51a0.

**Reply by Navid200** [reply]
Thanks so much for identifying those.

**Reply by jamorham** [reply]
Just report it to the anti-virus vendor as a false positive. There is nothing we can do. The pdf report flags things like using `wakelock->release()` as a sensitive api which makes no sense at all and implies they don't know much about android app internals.

**Reply by Puderzo** [reply]
I did flag this as a false positive, but I know from experience that the next version will probably be flagged again, as this appears to be an automated verdict. There are a few apps which fall into this same category of continuous false positives, Tasker being one of them, but Tasker I can understand as it really does use some quite "suspect" techniques. I have uploaded the verdict report for the non-flagged version below, if you want to have a look at any differences therein.

**Reply by jamorham** [reply]
I'm sure the manufacturer can whitelist any false positives. They just need to be complained at sufficiently to lift a finger to resolve the problem. The point here is that there is nothing wrong with xDrip, the faulty software is the anti-virus which cannot correctly identify benign software. This sort of thing happens all the time within the anti-virus community and daily they will be whitelisting things that they have incorrectly identified.

**Reply by Puderzo** [reply]
Indeed, I just got this in my e-mail:

```
WildFire Update Incorrect Verdict

SHA256: C49311E5DE38F3EE67A2F2FEB2D90D8C6087BBFFFB720FE83E250E67526E34D8
Received Time: 2024-01-24 15:46:54 (UTC)
Updated Time: 2024-01-29 16:12:43 (UTC)

After further review, the file was correctly categorized as benign, and no further action will be taken.
```

(along with 3 other xDrip+ versions I also flagged as false positives)

Unfortunately, I fully expect the next version to also be flagged, as this process appears to be automated on their end and seems to be erring on the side of cautious, resulting in these false positives. But since I'm running on alpha (the latest which is now correctly reported as benign) for my son's setup, this won't bother me until you guys update a new such version (at which point I will just report it again) and I can live with this either way :)

Thank you for taking the time to address this issue (and thank you for all the good work!).

**Reply by jamorham** [reply]
Eventually they may learn that the package is benign if you keep reporting false positives. Thanks for taking the time to report the issue and get it resolved from your end.

**Comment by tzachi-dar** [comment]
The main differences between the versions mentioned is carelink encryption.
The pdf attached says: "This sample was not found to contain any high-risk content during a pre-screening
analysis of the sample."
That said, can you also create a report for the older version so we can compare them?

**Reply by Puderzo** [reply]
Here's the verdict report for the older (non-flagged) version.
[1085f57adf2affe0513ecaf3dc885678e4361f89d4c93faf6afdf5da3fb923a9.pdf](https://github.com/NightscoutFoundation/xDrip/files/14085913/1085f57adf2affe0513ecaf3dc885678e4361f89d4c93faf6afdf5da3fb923a9.pdf)
